caravel/assets/spec/javascripts/sqllab/reducers_spec.js

```diff
@@ -25,4 +26,73 @@ describe('sqlLabReducer', () => {
       expect(newState.tabHistory[1]).to.eq(newState.queryEditors[1].id);
     });
   });
+  describe('Alerts', () => {
+    const state = Object.assign({}, initialState);
+    let newState;
+    it('should add one alert', () => {
+      newState = r.sqlLabReducer(state, actions.addAlert(alert));
+      expect(newState.alerts).to.have.lengthOf(1);
+    });
+    it('should remove one alert', () => {
+      newState = r.sqlLabReducer(newState, actions.removeAlert(newState.alerts[0]));
+      expect(newState.alerts).to.have.lengthOf(0);
+    });
+  });
+  describe('Query editors actions', () => {
+    const state = Object.assign({}, initialState);
+    let newState;
+    const defaultQueryEditor = state.queryEditors[0];
+    const qe = Object.assign({}, defaultQueryEditor);
+    qe.id = '2ed2';
+    it('should add a query editor', () => {
+      newState = r.sqlLabReducer(state, actions.addQueryEditor(qe));
+      expect(newState.queryEditors).to.have.lengthOf(2);
+    });
+    it('should remove a query editor', () => {
+      newState = r.sqlLabReducer(newState, actions.removeQueryEditor(qe));
+      expect(newState.queryEditors).to.have.lengthOf(1);
```

Comment:
we could check the length here, or we could be more explicit and check that `newState.queryEditors` doesn't contain a `qe` with the same id, so we know the correct `qe` was removed.